packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -146,16 +149,19 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     _setTween();
   }
 
-  void _handleTapDown(PointerDownEvent event) {
-    _animationController.animateTo(1.0, duration: kFadeOutDuration);
+  Future<Null> _handleTapDown(PointerDownEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(1.0, duration: kFadeOutDuration);
```

Comment:
imagine the user goes down-up-down-up-down.

The first down will start an animation to 1.0 (i assume you can `await null` with impunity?).
The next five callbacks all await that animation.
The animation ends at 1.0. The first callback to react is an up, it starts an animation to 0.0. Then the next callback starts an animation to 1.0 (which just stops the animation since it's already there), then the next one restarts it to 0.0, then we stop it again...

I guess it works, it just feels very wrong to be scheduling all these calls that stomp on each other.

Oh, also we need to check "mounted" after the await, in case the widget is gone and the controller has been disposed.

Also, we should fix this to use onTapDown/onTapUp/onTapCancel rather than a pointer listener.